frb_codegen/src/generator/rust/mod.rs

```diff
@@ -276,8 +286,31 @@ impl Generator {
             .collect::<Vec<_>>()
             .join("");
 
-        let code_call_inner_func = TypeRustGenerator::new(func.output.clone(), ir_file)
-            .wrap_obj(format!("{}({})", func.name, inner_func_params.join(", ")));
+        let code_call_inner_func = if !is_a_method {
+            TypeRustGenerator::new(func.output.clone(), ir_file).wrap_obj(format!(
+                "{}({})",
+                func.name,
+                inner_func_params.join(", ")
+            ))
+        } else {
+            let method_name = if is_method(&func.name) {
+                inner_func_params[0] = format!("&{}", inner_func_params[0]);
+                clear_method_marker(&func.name)
+            } else if is_static_method(&func.name) {
+                static_method_return_method_name(&func.name)
```

Comment:
Can we have even stronger type system for naming as follows (just rough draft, surely you can change the struct name etc):

```rust
struct FunctionName {
  actual_name: String,
  method_info: enum {
    NotMethod,
    StaticMethod {struct_name: String},
    NonStaticMethod {struct_name: String},
  }
}

impl FunctionName{
  fn serialize(&self) -> String {} // assemble it into a string, e.g. `actual_name___thestructname___static` etc
}

  fn deserialze(s: String) -> Self {} // the reverse
```

Reply:
I just tried this and ended up in the situation where sometimes I don't parse a string but I parse a IrFunc:

```rust
fn is_method_return_struct_name(func: &IrFunc) -> Option<String> {
        if Self::is_non_static_method(&func.name) {
            let input = func.inputs[0].clone();
            Some(input.name.to_string().to_case(Case::UpperCamel))
        } else if Self::is_static_method(&func.name) {
            Some(Self::static_method_return_struct_name(&func.name).to_case(Case::UpperCamel))
        } else {
            None
        }
    }
```

cause I need the input. Can we leave like this?

@fzyzcjy need your input before doing the rest, I'm almost in the end